clients/gtest/gesvd_gtest.cpp

```diff
@@ -0,0 +1,215 @@
+/* ************************************************************************
+ * Copyright 2020-2021 Advanced Micro Devices, Inc.
+ *
+ * ************************************************************************ */
+
+#include "testing_gesvd.hpp"
+
+using ::testing::Combine;
+using ::testing::TestWithParam;
+using ::testing::Values;
+using ::testing::ValuesIn;
+using namespace std;
+
+typedef std::tuple<vector<int>, vector<int>> gesvd_tuple;
+
+// each size_range vector is a {m, n, fa};
+// if fa = 0 then no fast algorithm is allowed
+// if fa = 1 fast algorithm is used when possible
+
+// each opt_range vector is a {lda, ldu, ldv, leftsv, rightsv};
+// if ldx = -1 then ldx < limit (invalid size)
+// if ldx = 0 then ldx = limit
+// if ldx = 1 then ldx > limit
+// if leftsv (rightsv) = 0 then overwrite singular vectors
+// if leftsv (rightsv) = 1 then compute singular vectors
+// if leftsv (rightsv) = 2 then compute all orthogonal matrix
+// if leftsv (rightsv) = 3 then no singular vectors are computed
+
+// case when m = -1, n = 1, and rightsv = leftsv = 3 will also execute the bad
+// arguments test (null handle, null pointers and invalid values)
+
+// for checkin_lapack tests
+const vector<vector<int>> size_range = {
+    // invalid
+    {-1, 1, 0},
+    {1, -1, 0},
+    // normal (valid) samples
+    {1, 1, 0},
+    {20, 20, 0},
+    {40, 30, 0},
+    {60, 30, 0},
+    {60, 30, 1}};
```

Comment:
You always call rocsolver_gesvd in-place, so this last test case is repeated; it could be removed.
(Not sure yet if it is a good idea to always call in place, though... see my other related comments)